### ISSUE TITLE
ENH: sklearn.model_selection.ShuffleSplit compatible wit pandas.MultiIndex

### ISSUE THREAD

**Issue by jeiglsperger** [pandas-dev/pandas#60552]
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I wish I could smoothly apply `sklearn.model_selection.ShuffleSplit `  with a `pandas.Dataframe` that has a `pandas.MultiIndex`. I tried to apply it, but it throws some weird `KeyError`.

### Feature Description

 This is my pseudo code how I would expect it to work:
```
from sklearn.model_selection import ShuffleSplit
from sktime.datatypes import get_examples
df = get_examples(mtype="pd-multiindex", as_scitype="Panel")[0]
splitter = ShuffleSplit(n_splits=3, random_state=42)
split = splitter.split(df.index.levels[0]) 
train_indexes = []
test_indexes = []
for train_index, test_index in split:
    train_indexes.append(train_index)
    test_indexes.append(test_index)
x_train, x_test = (df.loc[train_indexes[0]], df.loc[test_indexes[0]])
```

### Alternative Solutions

Currently I only to a train_test_split as described [here](https://gist.github.com/shaypal5/3e34e85bd89d65d4ac118daa9a42b174) without any cross validation.

### Additional Context

It would also be nice to extend this to the other split methods of sklearn.

**Comment by rhshadrach** [comment]
Thanks for the report.

> I tried to apply it, but it throws some weird `KeyError`.

Can you provide a fully reproducible example? In what you've provided, `df` and `x_train_val` are missing.

**Comment by jeiglsperger** [comment]
Sorry for not providing a MWE. I edited my post accordingly. In the MWE I used example data from `sktime`, for which the code works. There the error (certain indexes are not being found) is somehow due to my dataset. Either way, I now use the workaround to handle `ndarray` in the preprocessing and then convert them to `pd.DataFrame`
